services/console-proxy/server/src/com/cloud/consoleproxy/vnc/packet/server/RawRect.java

```diff
@@ -50,26 +52,27 @@ public void paint(BufferedImage image, Graphics2D graphics) {
 
         switch (dataBuf.getDataType()) {
 
-            case DataBuffer.TYPE_INT: {
-                // We chose RGB888 model, so Raster will use DataBufferInt type
-                DataBufferInt dataBuffer = (DataBufferInt)dataBuf;
+        case DataBuffer.TYPE_INT: {
+            // We chose RGB888 model, so Raster will use DataBufferInt type
+            DataBufferInt dataBuffer = (DataBufferInt)dataBuf;
 
-                int imageWidth = image.getWidth();
-                int imageHeight = image.getHeight();
+            int imageWidth = image.getWidth();
+            int imageHeight = image.getHeight();
 
-                // Paint rectangle directly on buffer, line by line
-                int[] imageBuffer = dataBuffer.getData();
-                for (int srcLine = 0, dstLine = y; srcLine < height && dstLine < imageHeight; srcLine++, dstLine++) {
-                    try {
-                        System.arraycopy(buf, srcLine * width, imageBuffer, x + dstLine * imageWidth, width);
-                    } catch (IndexOutOfBoundsException e) {
-                    }
+            // Paint rectangle directly on buffer, line by line
+            int[] imageBuffer = dataBuffer.getData();
+            for (int srcLine = 0, dstLine = y; srcLine < height && dstLine < imageHeight; srcLine++, dstLine++) {
+                try {
+                    System.arraycopy(buf, srcLine * width, imageBuffer, x + dstLine * imageWidth, width);
+                } catch (IndexOutOfBoundsException e) {
+                    s_logger.info("[ignored] buffer overflow!?!", e);
                 }
-                break;
             }
+            break;
```

Comment:
I guess it's OK. It looks like it would fix a bug (the for loop would only execute at most once before).

If you're OK with it, I'm OK with it.